course/include/isometry.h

```diff
@@ -0,0 +1,141 @@
+            x_ = *it++;
+            y_ = *it++;
+            z_ = *it;
+        }
+
+        // Operators
+        double& operator [] (const int);
+        const double& operator [] (const int) const;
+        Vector3 operator + (const Vector3&) const;
+        Vector3 operator - (const Vector3&) const;
+        Vector3 operator * (const Vector3&) const;
+        Vector3 operator * (const double&) const;
+        Vector3 operator / (const Vector3&) const;
+        Vector3& operator = (const Vector3&);
+        Vector3& operator += (const Vector3&);
+        Vector3& operator -= (const Vector3&);
+        Vector3& operator *= (const Vector3&);
+        Vector3& operator /= (const Vector3&);
+        bool operator == (const Vector3&) const;
+        bool operator != (const Vector3&) const;
+        bool operator == (const std::initializer_list<double>& list) const;
+        bool operator != (const std::initializer_list<double>& list) const;
+
+        friend Vector3 operator * (const double d, const Vector3& v) {
+            return Vector3(v.x_ * d, v.y_ * d, v.z_ * d);
+        }
+
+        friend std::ostream& operator << (std::ostream& os, const Vector3& v) {
+            os << std::string("(x: ") << v.x() << ", y: " << v.y() << ", z: " << v.z() << ")";
+            return os;
+        }
+
+        // Getters
+        double x() const {return x_;};
+        double y() const {return y_;};
+        double z() const {return z_;};
+
+        // Setters
+        double& x() {return x_;};
+        double& y() {return y_;};
+        double& z() {return z_;};
+
+        // Computations
+        double dot(const Vector3&) const;
+        double norm() const;
+        Vector3 cross(const Vector3&) const;
+
+        // Class constants
+        static const Vector3 kUnitX;
+        static const Vector3 kUnitY;
+        static const Vector3 kUnitZ;
+        static const Vector3 kZero;
+
+    private:
+        // Attributes
+        double x_;
+        double y_;
+        double z_;
+};
+
+class Matrix3{
+    public:
+        // Constructors
+        Matrix3(const Vector3& r1 = Vector3(), 
+                const Vector3& r2 = Vector3(), const Vector3& r3 = Vector3())
+                 : r1_{r1}, r2_{r2}, r3_{r3} {};
+        Matrix3(const Matrix3& m) : Matrix3(m.row(0), m.row(1), m.row(2)) {};
+        Matrix3(const std::initializer_list<double>& list);
+        Matrix3(const std::initializer_list<double>&,
+                const std::initializer_list<double>&,
+                const std::initializer_list<double>&);
+
+        // Operators
+        Vector3& operator [] (const int);
+        const Vector3& operator [] (const int) const;
+        Matrix3 operator + (const Matrix3&) const;
+        Matrix3 operator - (const Matrix3&) const;
+        Matrix3 operator * (const Matrix3&) const;
+        Matrix3 operator * (const double) const;
+        Matrix3 operator / (const Matrix3&) const;
+        Matrix3& operator = (const Matrix3&);
```

Comment:
For the sake of practice, also create a move assignment operator.

Reply:
I left you a question in the implementation of the move assignment